ibis/selectors.py

```diff
@@ -639,15 +672,48 @@ def expand_names(self, table: ir.Table) -> frozenset[str]:
         return frozenset(iterable)
 
 
-class Sliceable(Singleton):
+class Indexable(Singleton):
     def __getitem__(self, key: str | int | slice | Iterable[int | str]):
         if isinstance(key, slice):
             key = Slice(key.start, key.stop, key.step)
-        return ColumnSlice(key)
+        return ColumnIndex(key)
+
+
+index = Indexable()
+"""Select columns by index.
```

Comment:
This is annoying.

`index` is an "attribute" object in `griffe`. As such, it expects a docstring below the object definition (as done here). This works, the docstring is properly rendered. However, during runtime inspection the user won't see this docstring since it's not actually attached to anything.

Moving the docstring to class level on `Indexable` means an ipython/jupyter user can do `ibis.selectors.index?` and see the docstring. `quartodoc`/`griffe` won't properly find the docstring though unless you set `dynamic: true` (maybe a bug in `quartodoc`/`griffe`?). If you set `dynamic: true` though, the `Examples` aren't rendered properly as markdown (this is probably a bug in `quartodoc`). Manually wrapping in the docstring works, but is a pain.

For now I've opted to do the easy thing and just get our docs rendering.

Reply:
cc @machow for visibility